### ISSUE TITLE
Worms WMD (327030)

### ISSUE THREAD

**Issue by Protonator** [ValveSoftware/Proton#2953]
# Compatibility Report
- Name of the game with compatibility issues: Worms WMD
- Steam AppID of the game: 327030

## System Information
- GPU: GeForce GTX 1080
- OS: Manjaro Cinnamon
- Driver/LLVM version: NVIDIA 430.26, Mesa 19.1.3
- Kernel version: 5.2.4
- Proton version: 4.2 and 4.11
- [Full system information](https://gist.github.com/Protonator/cc65853478e3fd3e84414a9b757102cf#file-sysinfo-txt)

## I confirm:
- [X] that I haven't found an existing compatibility report for this game.
- [X] that I have checked whether there are updates for my system available.

[steam-327030.log](https://github.com/ValveSoftware/Proton/files/3464868/steam-327030.log)

## Symptoms 

According to the title screen, Worms WMD servers are not reachable, thus features like the Ranked mode, the global chat and the Ranked stats doesn't work.

## Reproduction

1. Launch the game with Proton
2. Open global chat, Ranked stats or Ranked mode. None of these features are usable.

## Working features

Unranked multiplayer, private chats, solo mode and all other game or menu options are fully functional via Proton, beside (known Proton) problems with [Steam overlay](https://github.com/ValveSoftware/Proton/issues/168), [special characters](https://github.com/ValveSoftware/steam-for-linux/issues/2122) and [video playback](https://github.com/ValveSoftware/Proton/issues/1464) (WMV). Under Windows 10 on the same system all of the above problems doesn't exist.

**Comment by AwesamLinux** [comment]
**System Information**
- GPU: RX VEGA 56
- GPU Driver: Mesa ACO 19.1.0+aco+git1564747450-10690f8d9e6-1bionic1 (Valves PPA)
- Kernel version:  5.2.5-050205-lowlatency
- Distro version: Linux Mint 19.2 Tina (64 bit) (Cinnamon Desktop)
- Proton version: 4.11-1

**Compatibility Report**

I can confirm that videos do not play (just black screens, have to wait/press keys to continue). Other than that offline single player appears to work fine (did not try online multiplayer).

#1464 Videos are: WMV, VC-1 + WMA
Looks like a quartz game, In the logs I see `26602.765:0027:0028:err:quartz:GetClassMediaFile Media class not found`

[media_info.txt](https://github.com/ValveSoftware/Proton/files/3465283/media_info.txt)
[steam-327030-d9vk-proton-4.11-1b.log](https://github.com/ValveSoftware/Proton/files/3465284/steam-327030-d9vk-proton-4.11-1b.log)
[steam-327030-proton-4.2-9.log](https://github.com/ValveSoftware/Proton/files/3465285/steam-327030-proton-4.2-9.log)
[steam-327030-proton-4.11-1b.log](https://github.com/ValveSoftware/Proton/files/3465286/steam-327030-proton-4.11-1b.log)
[system_info.txt](https://github.com/ValveSoftware/Proton/files/3465287/system_info.txt)

(Note that this game is listed as a Linux native on Steam, was the Linux version removed at some point?)

**Comment by Protonator** [comment]
> (Note that this game is listed as a Linux native on Steam, was the Linux version removed at some point?)

No, but the native Linux version is not well tested and doesn't work out of the box with certain distributions other than Ubuntu. I haven't been able to get it work under Manjaro with the circulating workarounds. But even if it works, you have no functioning Steam overlay there, too. That's the reason why I switched to Proton for the game.

**Comment by Pigpog** [comment]
> I haven't been able to get it work under Manjaro with the circulating workarounds.

I was able to play Worms on Manjaro by following [this guide](https://steamcommunity.com/app/327030/discussions/2/133257959065155871/).

Hope this helps.

**Comment by Protonator** [comment]
Yes, I tried these workarounds (and others). Nothing helped. But thanks anyway.

**Comment by Protonator** [comment]
I just tried it again with the guide and recognized that I formerly haven't adjusted the path names in the script. Now the native Linux version works here, too. I also had to install the packages lib32-libcurl-gnutls and wavpack and symlink the libcurl-gnutls. So thank you very much for the tip!

By the way: Other than via Proton, video playback is working in the native Linux version thanks to wavpack. And I can use special characters like vowels there, too. Ranked multiplayer and Steam overlay are broken in both variants, though. And the game only has a 60 Hz option (other than with Proton, where I can set and use up to 144 Hz), which leads to a hardly playable mouse lag. I will switch back to Proton now.

Edit: As a workaround I let Steam switch to US keyboard layout when the game starts, so I can at least use "?" in the ingame chat. After I exit the game it switches automatically back to my German layout. This is the specific game launch option:

`gamemoderun setxkbmap us && strangle 140 %command%; setxkbmap de`

As you can see, it involves the [Feral gamemode](https://github.com/FeralInteractive/gamemode) and the fps limiter [libstrangle](https://github.com/milaq/libstrangle), too.

**Comment by pinpox** [comment]
Any workaround to get ranked multiplayer working on Linux?

**Comment by pablodav** [comment]
I changed the compatibility to Proton 5.13 and finally was able to open the game.

On flatpak steam works with latest community build of proton 6.3 20210506.

Videos does not play, but works.

Error when playing video:

```shell
(wine:6569): GStreamer-WARNING **: 20:48:52.075: Failed to load plugin '/home/pablo/.local/share/Steam/steamapps/common/Proton 5.13/dist/lib64/gstreamer-1.0/libgstautodetect.so': /home/pablo/.local/share/Steam/steamapps/common/Proton 5.13/dist/lib64/gstreamer-1.0/libgstautodetect.so: clase ELF errónea: ELFCLASS64

(wine:6569): GStreamer-WARNING **: 20:48:52.075: Failed to load plugin '/home/pablo/.local/share/Steam/steamapps/common/Proton 5.13/dist/lib64/gstreamer-1.0/libgstapp.so': /home/pablo/.local/share/Steam/steamapps/common/Proton 5.13/dist/lib64/gstreamer-1.0/libgstapp.so: clase ELF errónea: ELFCLASS64

(wine:6569): GStreamer-WARNING **: 20:48:52.075: Failed to load plugin '/home/pablo/.local/share/Steam/steamapps/common/Proton 5.13/dist/lib64/gstreamer-1.0/libgstmpg123.so': /home/pablo/.local/share/Steam/steamapps/common/Proton 5.13/dist/lib64/gstreamer-1.0/libgstmpg123.so: clase ELF errónea: ELFCLASS64

(wine:6569): GStreamer-WARNING **: 20:48:52.075: Failed to load plugin '/home/pablo/.local/share/Steam/steamapps/common/Proton 5.13/dist/lib64/gstreamer-1.0/libgstvideofilter.so': /home/pablo/.local/share/Steam/steamapps/common/Proton 5.13/dist/lib64/gstreamer-1.0/libgstvideofilter.so: clase ELF errónea: ELFCLASS64
```

**Comment by gituser823** [comment]
The game runs with GE-Proton7-10.
But I have no sound.

Any way to fix that?

I am on Mint

**Comment by blamedrop** [comment]
Just tried to run `Worms W.M.D` (327030) with Proton:

- `Proton 9.0-4`
- `Proton Experimental [bleeding-edge]`
- `Proton Experimental`

with no launch options, game just exits right away, no window shows up (just the Steam button goes through `PLAY`→`CANCEL`→`STOP`→`PLAY` states).

There was nothing useful in Steam output and Proton log.
I've decided to remove `~/.local/share/Steam/steamapps/compatdata/327030` directory.
The **game then started to work just fine**, duh!

Not sure what was that about, sharing it just as a heads up for anyone that would have similar problem. Unfortunately I've deleted that `compatdata` directory for real already and can't check if there were any logs :/ It also could be an issue of Steam Cloud synchronizing files including graphical settings from Windows PC that I've used to play it in the meantime. Anyway, it works now.

One issue tho - Steam Overlay is not showing up.
Running it via `gamescope` fixes that, so everything is fine!

**Comment by Pigpog** [comment]
In my experience, deleting compatdata has been necessary for all games when switching Proton versions
